### PR TITLE
Use precice develop for tests

### PR DESCRIPTION
**Pull request by erikscheurer**
This PR modifies the tests to build precice develop instead of using the latest precice docker container as discussed in #47.

### CODE REVIEW

**Comment by erikscheurer** [comment]
At the moment, only the build test uses precice develop, this can easily be added, but may be open for debate as the [python bindings](https://github.com/precice/python-bindings/blob/71e7a6e38af3418192b3d8f65006bc8f97a1d4dd/.github/workflows/build-and-test.yml#L134) to my understanding also use the docker container to test the solver dummies. The solverdummies would be the only test affected by this, as the others use a dummy solverinterface.

**Comment by erikscheurer** [comment]
We are now facing the issue, that the tests would run fine, but the versions of julia are outdated in a way, that leads the version of `libstdc++` being newer in the container that we use (`ubuntu-latest`) than the ones shipped with julia.
This error has been discussed in #44 and is now included in the julia bindings readme. It could be solved by preloading the system libraries, or by upgrading to newer julia versions compared to the system.

**Comment by IshaanDesai** [comment]
> We are now facing the issue, that the tests would run fine, but the versions of julia are outdated in a way, that leads the version of `libstdc++` being newer in the container that we use (`ubuntu-latest`) than the ones shipped with julia. This error has been discussed in #44 and is now included in the julia bindings readme. It could be solved by preloading the system libraries, or by upgrading to newer julia versions compared to the system.

I do not fully understand what can be a solution here. Are you suggesting that we drop support for older Julia versions? The test failures seem to not be related to Julia versions, because some tests for `1.6.0` pass and others fail.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -34,19 +33,23 @@ jobs:
       - name: Install Julia
         uses: julia-actions/setup-julia@v1
         with:
-          version: ${{ matrix.julia-version }}     
-        
-      - uses: actions/checkout@v2
+          version: ${{ matrix.julia-version }}  
+   
+      - uses: precice/setup-precice-action@testing # needs to be changed to main after merge
```
Comment (MakisH) [inline .github/workflows/build-and-test.yml:38]: So, actions are just sourced from any repository on GitHub? That's nice. We can then use it however we want and then it depends if and how much we advertise it to others.

**Comment by erikscheurer** [comment]
These changes break the tests on develop, but this should be fixed with #47.
Due to older LTS versions of Julia, we have to preload `libstdc++` and `libcurl` as described in #44 and now the [README](https://github.com/precice/PreCICE.jl#incompatible-version-of-libstdc).

**Review by IshaanDesai — APPROVED** [review]
Looks good :+1: please squash and merge.

**Pull request merged**